modules/dns/main.tf

```diff
@@ -21,3 +21,20 @@ resource "ibm_cis_dns_record" "dns_records" {
   data      = ((each.value.data != null) && (each.value.type == "SRV")) ? merge(each.value.data, { "name" : lookup(each.value, "name", "") }) : lookup(each.value, "data", null)
   proxied   = lookup(each.value, "proxied", false)
 }
+
+##############################################################################
+# Import DNS records from a file
+##############################################################################
+resource "ibm_cis_dns_records_import" "dns_record" {
+  count      = (length(var.base64_encoded_dns_file) != 0 || var.dns_file != null) ? 1 : 0
+  depends_on = [local_file.dns_record_file]
+  cis_id     = var.cis_instance_id
+  domain_id  = var.domain_id
+  file       = length(var.base64_encoded_dns_file) != 0 ? local_file.dns_record_file[0].filename : var.dns_file
+}
```

Comment:
The code does not stop me providing dns_file.txt and encoded string both. But it does not work in that way. If both are provided, then it is working only for base64 encoded one not for text file.

Also, the file name `dns_records.txt` is hardcoded in `local_file resource` , I think it can be prefixed by instance id / domain id /...to avoid any duplication (if any).